modules/alloy/src/main/ts/ephox/alloy/positioning/view/Callouts.ts

```diff
@@ -30,7 +32,9 @@ const layout = (anchorBox: AnchorBox, element: SugarElement, bubbles: Bubble, op
 const setClasses = (element: SugarElement, decision: RepositionDecision): void => {
   const classInfo = decision.classes;
   Classes.remove(element, classInfo.off);
+  Classes.remove(element, LayoutLabels.all);
   Classes.add(element, classInfo.on);
+  Classes.add(element, [ decision.label ]);
```

Comment:
Should this be a `data-mce-{name}` attribute instead of a CSS class? It looks like we're adding this in purely as a way to attach state to an element, and so using CSS classes would introduce a lot of risks (accidental interop with the integrator's stylesheets, accidentally ending up with multiple incompatible classes on the same element) without any benefit (we're not targeting these elements with our own CSS, and semantically it isn't the best way to represent this state anyway).

Also, regardless of which way we attach these things to elements, I'd recommend putting a single getter / setter into the `LayoutLabels` file instead of duplicating the "where on the element is this state stored" logic.

Reply:
That might be a posibility, yeah, we could do that. Would probably remove the need for the `all` as well, if we can just remove the singular attribute.